crates/bevy_pbr/src/atmosphere/functions.wgsl

```diff
@@ -305,9 +309,19 @@ fn max_atmosphere_distance(r: f32, mu: f32) -> f32 {
     return mix(t_top, t_bottom, f32(hits));
 }
 
-/// Assuming y=0 is the planet ground, returns the view radius in meters
-fn view_radius() -> f32 {
-    return max(view.world_position.y * settings.scene_units_to_m, EPSILON) + atmosphere.bottom_radius;
+/// Returns the observer's position in the atmosphere
+fn get_view_position(ray_dir: vec3<f32>) -> vec3<f32> {
+    var world_pos = view.world_position * settings.scene_units_to_m + vec3(0.0, atmosphere.bottom_radius, 0.0);
+    
+    // move the world position forward to the first intersection with the ground
```

Comment:
so if you're under the ground, rays pointing down go all the way to the other side of the earth?

Reply:
Yes thats the only sensible way i could resolve this case. Ideas and suggestions are welcome though

Reply:
What do rays pointing down do when you're above ground?

Reply:
They start at the camera and end at the ground. Or if outside the atmosphere then they start at the top of the atmosphere.